### PR TITLE
[C-1810] Block more screens when offline

### PR DESCRIPTION
**Pull request by amendelsohn**
### Description

Shows the offline indicator instead of failing to show content that requires connectivity.

Recommend review with hide whitespace changes.

### Dragons

Might be better off with a blanket solution for all screens. We could pass in an `offline={true}` prop for those that don't require it.

### CODE REVIEW

File: packages/mobile/src/screens/chats-screen/ChatsScreen.tsx [hunk 1]
```diff
@@ -24,18 +24,20 @@ export const ChatsScreen = () => {
...
-          return (
-            <View key={message.message_id}>
-              <Text>{message.sender_user_id} says:</Text>
-              <Text>{message.message}</Text>
-              <Text>{message.created_at}</Text>
-            </View>
-          )
-        })}
-      </View>
+      <ScreenContent>
```
Comment (dylanjeffers) [inline packages/mobile/src/screens/chats-screen/ChatsScreen.tsx:27]: yeah could we just make the offline thing an option for screen? and default it to "not offline capable", and just the few screens that need it, we go `<Screen allowOffline />` or something?

**Review by dylanjeffers — COMMENTED** [review]
If we are in a big rush, maybe go forward with current impl, but imo moving this to screen would result in way fewer changes and less cognitive complexity potentially?

File: packages/mobile/src/screens/chats-screen/ChatsScreen.tsx (hunk 1 above)
Comment (amendelsohn) [reply packages/mobile/src/screens/chats-screen/ChatsScreen.tsx:27]: Yes, I think that's worth doing. I'll circle back to this tonight and do it that way.

**Review by dylanjeffers — APPROVED** [review]
based on chats, the move is to wrap all screen "content" in "ScreenContent"!

**Pull request merged**